contrib/win32/win32compat/console.c

```diff
@@ -1090,14 +1094,19 @@ ConMoveVisibleWindow(int offset)
 			for (int i = 0; i < offset; i++)
 				ConScrollDown(0, consoleInfo.dwSize.Y - 1);
 
-			if (GetConsoleScreenBufferInfo(hOutputConsole, &consoleInfo))
-				memcpy(&visibleWindowRect, &consoleInfo.srWindow, sizeof(visibleWindowRect));
-			else {
+			if (GetConsoleScreenBufferInfo(hOutputConsole, &consoleInfo) == FALSE) {			
 				error("GetConsoleScreenBufferInfo failed with %d", GetLastError());
 				return;
 			}
+			if (memcpy_s(&visibleWindowRect, sizeof(visibleWindowRect), &consoleInfo.srWindow, sizeof(visibleWindowRect))) {
+				error("memcpy_s failed: %d.", errno);
+				return;
+			}
 		} else {
-			memcpy(&visibleWindowRect, &consoleInfo.srWindow, sizeof(visibleWindowRect));
+			if (memcpy_s(&visibleWindowRect, sizeof(visibleWindowRect), &consoleInfo.srWindow, sizeof(visibleWindowRect))) {
```

Comment:
If execution is allowed to continue, these functions return EINVAL and set errno to EINVAL.